internal/interfaces/grpc/handler/operator.go

```diff
@@ -1362,3 +1391,24 @@ func toUtxoInfoList(list []application.UtxoInfo) []*pb.UtxoInfo {
 	}
 	return res
 }
+
+func parseTimeRange(timeRange *pb.TimeRange) application.TimeRange {
+	var predefinedPeriod *application.PredefinedPeriod
+	if timeRange.GetPredefinedPeriod() > pb.PredefinedPeriod_NULL {
+		pp := application.PredefinedPeriod(timeRange.GetPredefinedPeriod())
+		predefinedPeriod = &pp
+	}
+
+	var customPeriod *application.CustomPeriod
+	if timeRange.GetCustomPeriod() != nil {
+		customPeriod = &application.CustomPeriod{
+			StartDate: timeRange.GetCustomPeriod().GetStartDate(),
+			EndDate:   timeRange.GetCustomPeriod().GetEndDate(),
+		}
+	}
+
+	return application.TimeRange{
+		PredefinedPeriod: predefinedPeriod,
+		CustomPeriod:     customPeriod,
+	}
+}
```

Comment:
If we exported the validation method of `application.TimeRange`, this would become:
```suggestion
func parseTimeRange(timeRange *pb.TimeRange) (*application.TimeRange, error) {
	var predefinedPeriod *application.PredefinedPeriod
	if timeRange.GetPredefinedPeriod() > pb.PredefinedPeriod_NULL {
		pp := application.PredefinedPeriod(timeRange.GetPredefinedPeriod())
		predefinedPeriod = &pp
	}

	var customPeriod *application.CustomPeriod
	if timeRange.GetCustomPeriod() != nil {
		customPeriod = &application.CustomPeriod{
			StartDate: timeRange.GetCustomPeriod().GetStartDate(),
			EndDate:   timeRange.GetCustomPeriod().GetEndDate(),
		}
	}

	tr := &application.TimeRange{
		PredefinedPeriod: predefinedPeriod,
		CustomPeriod:     customPeriod,
	}
	if err := tr.Validate(); err != nil {
	   return nil, err
	}
	return tr, nil
}
```

Reply:
Done